data/binding/gen.go

```diff
@@ -66,12 +66,19 @@ func (b *bound{{ .Name }}) Get() ({{ .Type }}, error) {
 }
 
 func (b *bound{{ .Name }}) Reload() error {
-	return b.Set(*b.val)
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	b.trigger()
```

Comment:
This seems to do the opposite of your PR title - it will call the trigger irrespective of the change. 
Looking more does it even set the value?

Reply:
> it will call the trigger irrespective of the change.

@andydotxyz Yes, I am aware of it. But `Reload` is part only of `ExternalXxx`, then directly trigger would be the only thing we can do, as it would already have the `b.Val` set by reference? 🤔. From my perspective `Reload` is like a force trigger, isn't it?

Reply:
`Reload` is not like forcing trigger. It is ensuring data is up to date. 
If you consider a `[]string` bind with 10000 items we only want to trigger refresh on the items that have changed - to save 99997 callbacks.
We should be consistent and having internal and external data behave differently is not good.
One reason this was so contentious was that to correctly handle "if-changed" for all types would requiring storing a duplicate for comparison on the types where the data is external.

Reply:
> Reload is not like forcing trigger. It is ensuring data is up to date.

But `ExternalXxx` versions would always ensure data is up to date, as you are saving its reference. So the only thing `Reload` can do is trigger the listener's callback to report that change. If user does not want to trigger unnecessarily all the listeners (assuming the change of the external variable is the same as before), then they can use `Set` function.

> If you consider a []string bind

`DataLists` work different internally I think, this change is only for `DataItems`.

> We should be consistent and having internal and external data behave differently is not good.

Well considering `Reload` is only present in `ExternalXxx` variants, I don't think they behave different from internal ones with this change, do they?